src/libmongoc/tests/unified/runner.c

```diff
@@ -592,8 +600,13 @@ test_setup_initial_data (test_t *test, bson_error_t *error)
       bool ret = false;
 
       bson_iter_bson (&initial_data_iter, &collection_data);
-      database_name = bson_lookup_utf8 (&collection_data, "databaseName");
-      collection_name = bson_lookup_utf8 (&collection_data, "collectionName");
+      parser = bson_parser_new ();
+      bson_parser_utf8 (parser, "databaseName", &database_name);
+      bson_parser_utf8 (parser, "collectionName", &collection_name);
+      bson_parser_array_optional (parser, "documents", &documents);
```

Comment:
Isn't this required? From [here](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#collectiondata),

> documents: Required array of objects. List of documents corresponding to the contents of the collection. This list may be empty.

Reply:
Good catch. The check below was wrong. The array always exists. If it is empty, we need to create the collection explicitly.